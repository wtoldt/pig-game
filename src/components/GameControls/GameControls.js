import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Die from './Die/Die';

import './GameControls.scss';

export default class GameControls extends Component {

  static propTypes = {
    onGameReset: PropTypes.func.isRequired,
    onRoll: PropTypes.func.isRequired,
    onHold: PropTypes.func.isRequired,
    dieValue: PropTypes.number,
    disabled: PropTypes.bool
  };

  static defaultProps = {
    dieValue: 1,
    disabled: false
  };

  render() {
    const { onGameReset, onRoll, onHold, dieValue, disabled } = this.props;
    const classnames = classNames('pg-game-controls');
    return (
      <div className={classnames}>
        <div className="reset-btn-container">
          <button onClick={onGameReset}>Reset Game</button>
        </div>
        <div className="die-container">
          <Die value={dieValue} />
        </div>
        <div className="roll-btn-container">
          <button onClick={onRoll} disabled={disabled}>Roll</button>
        </div>
        <div className="hold-btn-container">
          <button onClick={onHold} disabled={disabled}>Hold</button>
        </div>
      </div>
    );
  }
}