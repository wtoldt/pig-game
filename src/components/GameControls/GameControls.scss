@import '../../styles/consts.scss';

.pg-game-controls {
  display: flex;
  flex-direction: column;
  width: 100px;
  background-color: mistyrose;
  align-items: center;

  .die-container {
    flex: 4;
    display: flex;
    align-items: center;
  }
}