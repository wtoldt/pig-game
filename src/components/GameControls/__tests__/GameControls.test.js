import { render } from '@testing-library/react';
import GameControls from '../GameControls';

const defaultProps = {
  onGameReset: () => {},
  onRoll: () => {},
  onHold: () => {},
  dieValue: () => {},
  disabled: false,
  dieValue: 1
};

function getWrapper(props={}) {
  return render(
    <GameControls {...defaultProps} {...props} />
  );
}

describe('GameControls Component', () => {

  describe('Snapshot testing', () => {
    it('Default snapshot', () => {
      const { container } = getWrapper();
      expect(container).toMatchSnapshot();
    });

    it('Disabled snapshot', () => {
      const { container } = getWrapper({ disabled: true });
      expect(container).toMatchSnapshot();
    });

  });
});
