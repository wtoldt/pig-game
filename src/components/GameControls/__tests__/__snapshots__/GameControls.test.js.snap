// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameControls Component Snapshot testing Default snapshot 1`] = `
<div>
  <div
    class="pg-game-controls"
  >
    <div
      class="reset-btn-container"
    >
      <button>
        Reset Game
      </button>
    </div>
    <div
      class="die-container"
    >
      <div
        class="pg-die"
      >
        <div
          class="row"
        >
          <div
            class="cell"
            id="c1"
          >
             
          </div>
          <div
            class="cell"
            id="c2"
          >
             
          </div>
          <div
            class="cell"
            id="c3"
          >
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="cell"
            id="c4"
          >
             
          </div>
          <div
            class="cell"
            id="c5"
          >
            üêΩ
          </div>
          <div
            class="cell"
            id="c6"
          >
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="cell"
            id="c7"
          >
             
          </div>
          <div
            class="cell"
            id="c8"
          >
             
          </div>
          <div
            class="cell"
            id="c9"
          >
             
          </div>
        </div>
      </div>
    </div>
    <div
      class="roll-btn-container"
    >
      <button>
        Roll
      </button>
    </div>
    <div
      class="hold-btn-container"
    >
      <button>
        Hold
      </button>
    </div>
  </div>
</div>
`;

exports[`GameControls Component Snapshot testing Disabled snapshot 1`] = `
<div>
  <div
    class="pg-game-controls"
  >
    <div
      class="reset-btn-container"
    >
      <button>
        Reset Game
      </button>
    </div>
    <div
      class="die-container"
    >
      <div
        class="pg-die"
      >
        <div
          class="row"
        >
          <div
            class="cell"
            id="c1"
          >
             
          </div>
          <div
            class="cell"
            id="c2"
          >
             
          </div>
          <div
            class="cell"
            id="c3"
          >
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="cell"
            id="c4"
          >
             
          </div>
          <div
            class="cell"
            id="c5"
          >
            üêΩ
          </div>
          <div
            class="cell"
            id="c6"
          >
             
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="cell"
            id="c7"
          >
             
          </div>
          <div
            class="cell"
            id="c8"
          >
             
          </div>
          <div
            class="cell"
            id="c9"
          >
             
          </div>
        </div>
      </div>
    </div>
    <div
      class="roll-btn-container"
    >
      <button
        disabled=""
      >
        Roll
      </button>
    </div>
    <div
      class="hold-btn-container"
    >
      <button
        disabled=""
      >
        Hold
      </button>
    </div>
  </div>
</div>
`;
